dashboard/lib/next-auth/dynamodb-adapter.ts

```diff
@@ -261,6 +263,23 @@ export async function createSession(session: {
   }
 }
 
+// Because next auth sucks
+export async function getSessionFromCookie(
+  req:
+    | NextApiRequest
+    | {
+        cookies: NextApiRequestCookies
+      },
```

Comment:
I'm not sure how to abstract this fragment type out -- Next API handlers use a different request than GetServeSideProps does.